homeassistant/components/denonavr/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Support for Denon AVR additional sensors."""
+
+from typing import Any
+
+from denonavr import DenonAVR
+from denonavr.const import POWER_ON
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import StateType
+
+from .device import DenonDeviceEntity
+
+ATTR_ZONE = "zone"
+
+SENSOR_BASS = "bass"
+SENSOR_BASS_LEVEL = "bass_level"
+SENSOR_DYNAMIC_EQ = "dynamic_eq"
+SENSOR_DYNAMIC_VOLUME = "dynamic_volume"
+SENSOR_MULTI_EQ = "multi_eq"
+SENSOR_REFERENCE_LEVEL_OFFSET = "reference_level_offset"
+SENSOR_TONE_CONTROL_ADJUST = "tone_control_adjust"
+SENSOR_TONE_CONTROL_STATUS = "tone_control_status"
+SENSOR_TREBLE = "treble"
+SENSOR_TREBLE_LEVEL = "treble_level"
+
+SENSOR_ENTITIES = {
+    SENSOR_BASS: "Bass",
+    SENSOR_BASS_LEVEL: "Bass Level",
+    SENSOR_DYNAMIC_EQ: "Dynamic EQ",
+    SENSOR_DYNAMIC_VOLUME: "Dynamic Volume",
+    SENSOR_MULTI_EQ: "Multi EQ",
+    SENSOR_REFERENCE_LEVEL_OFFSET: "Reference Level Offset",
+    SENSOR_TONE_CONTROL_ADJUST: "Tone Control Adjust",
+    SENSOR_TONE_CONTROL_STATUS: "Tone Control Status",
+    SENSOR_TREBLE: "Treble",
+    SENSOR_TREBLE_LEVEL: "Treble Level",
+}
+
+
+class DenonSensor(DenonDeviceEntity, SensorEntity):
+    """Representation of a Denon Sensor Device."""
+
+    def __init__(
+        self,
+        receiver: DenonAVR,
+        unique_id: str,
+        config_entry: ConfigEntry,
+        attribute: str,
+    ) -> None:
+        """Initialize the device."""
+        super().__init__(receiver, unique_id, config_entry)
+
+        self._attribute = attribute
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return device specific state attributes."""
+        return {ATTR_ZONE: getattr(self._receiver, ATTR_ZONE, None)}
```

Comment:
Can you explain how media player entities are created at the moment? I understood that we create one per zone, is that correct?

Reply:
Yes, one per zone. After the change we will have 11 per zone. The zone attribute is that you are able to see which sensor entity belongs to which media player (main) entity.

Reply:
But you can already see to which receiver it belongs, so how much added value do these extra state attributes add?

Fyi, we're more hesitant to add new extra attributes, because they usually can be extra sensors or are unneeded, but still take a lot of space in the database.

Reply:
Tbh., I would have problems to see the relationship between the media player entity and the sensor entities without the `Zone` attribute.
I can assume that there is a relationship based on the suffixes of the entity names (no suffix, `_2`, `_3`), but this relationship is gone when I start renaming the entities. That's why I added this property.

However, I checked the properties and all the sensors belong to properties which are device settings and not zonal settings. Thus, we don't need to create them multiple time and will have 10 sensor entities + one media player entity per zone only.

Reply:
Is every media player its own device?

Don't I then have a device called "living room" and "bedroom"?

Reply:
Nope, a device represents a physical receiver which can have up to 3 zones. Each zone is represented by a media player.
It is easy to figure out a relationship between a device and its entities, but it is hard to figure out relationships between different entities of the same device.
Anyway, the relationship between entities is a theoretical discussion now, because there is only one sensor entity per device left.

Reply:
I am more talking about the relationship between devices and the sensor entities. If the device is called "Bedroom" and you attach these entities, it would prepend the devicename, avoiding the `_1` and `_2`. It would look like `sensor.bedroom_something`. This would also rename when you rename the device, it will ask you if you also want to rename the entities attached to it.

But looking at the sensor setup code, you only setup the sensors for one entry and not per zone.

Reply:
> But looking at the sensor setup code, you only setup the sensors for one entry and not per zone.

Yes, that's what I said in my second message and did with https://github.com/home-assistant/core/pull/112857/commits/04173b884800a33d0652cbf31ff5192431d9bd84

> However, I checked the properties and all the sensors belong to properties which are device settings and not zonal settings. Thus, we don't need to create them multiple time and will have 10 sensor entities + one media player entity per zone only.